src/Microsoft.ML.Data/Transforms/TermTransformImpl.cs

```diff
@@ -454,8 +451,26 @@ private enum MapType : byte
         /// These are the immutable and serializable analogs to the <see cref="Builder"/> used in
         /// training.
         /// </summary>
-        private abstract class TermMap
+        internal abstract class TermMap
         {
+            private static volatile MemoryStreamPool _codecFactoryPool;
+            private static volatile CodecFactory _codecFactory;
+            private static object _factoryLock = new object();
+
+            private static CodecFactory CodecFactory(IHostEnvironment env)
```

Comment:
>private static [](start = 12, length = 14)

By making this object *static*, we are effectively making the environment and host composed out of that environment static, which we have indicated many, many times we do not want to do. If this `CodecFactory` must be defined with an environment, then it is a mistake to make instances of it static, by that token.

Reply:
moved back to TermTransform

---
In reply to: [214464720](https://github.com/dotnet/machinelearning/pull/759#discussion_r214464720) [](ancestors = 214464720)